Nix env

cpcloud opened the pull request:
Testing

pep8speaks commented:
Hello @cpcloud! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:







* In the file [`setup.py`](https://github.com/cpcloud/ibis/blob/e61338a9faa61fa16bffdd7304ab44ec84bc20aa/setup.py):

> [Line 130:80](https://github.com/cpcloud/ibis/blob/e61338a9faa61fa16bffdd7304ab44ec84bc20aa/setup.py#L130): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (2082 > 79 characters)



##### Comment last updated at 2021-09-16 17:07:42 UTC

cpcloud committed:
refactor(ci): refactor CI and development process

cpcloud committed:
refactor(ci): move to poetry

cpcloud committed:
ci: temporarily add pure-sasl to impala conda dependencies (#2987)

Lunaticus7 committed:
fix: fix literal of Nullable in Clickhouse (#2985)

* support literal None as Nullable column

* add test

* change to public api

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

cpcloud committed:
test: don't suppress test start info (#2986)

Lunaticus7 committed:
fix order of limit and offset param (#2984)

cpcloud committed:
test: fix incorrect connection pool test (#2990)

cpcloud committed:
fix: fix cursor accounting races (#2991)

cpcloud committed:
docs: add missing release notes for #2984 and #2985 (#2992)

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#2994)

updates:
- [github.com/psf/black: 21.8b0 → 21.9b0](https://github.com/psf/black/compare/21.8b0...21.9b0)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

cpcloud committed:
test: fix incorrect assumption about GC in impala test (#2993)

emilyreff7 committed:
Call aggregation function directly for pandas Summarize (#2995)

Co-authored-by: emilyreff <emilyreff@email.arizona.edu>

cpcloud committed:
ci: remove pure-sasl dependency (#2989)

cpcloud committed:
test: ensure that python ci/datamgr.py hdf5 produces data (#2999)

datapythonista committed:
CLN: Deprecate and standardize exists_table (#2905)

* CLN: Deprecate and standardize exists_table

* Fixing tests

* Fixing test

* Fix for database parameter

* Removing hack after list_table clean up

* Adding exists_table implementation to impala client

* Remove custom impala method

* Improving failing test output

* Replacing in tests exists_table by in list_tables

* Allowing full qualified table names

* Fixing tests

* Fixing test

* Fix typo

* Reverting exists_table full qualified handling

* Unqualifying table names to check if they exist

* Fix typo

* Adding verbosity to pytest

* Adding debug info

* Testing if adding previous parameters to list_table returns the missing table

* Removing broken test, adding release note and test the future warning

* Fixing file tests

* Restore failing test

* Considering a table exists using list_tables with like param

cpcloud committed:
chore: move to string arg

cpcloud committed:
ci: use a bit of nix in ci: docs, benchmarks and linting

cpcloud committed:
ci: make steps explicitly depend on commitlint

cpcloud committed:
ci: split asv into setup and execution

cpcloud committed:
ci: add release step
